metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableOpWrapper.java

```diff
@@ -45,7 +45,9 @@
     public Map<String, ScanSummary.PartitionMetrics> getPartitionMetricsMap(final Table icebergTable,
                                                                      @Nullable final Expression filter) {
         return (filter != null)
-            ? ScanSummary.of(icebergTable.newScan().filter(filter)).build()
+            ? ScanSummary.of(icebergTable.newScan().filter(filter))
+                .limit(connectorContext.getConfig().getIcebergTableSummaryFetchSize())
+                .build()
             :
```

Comment:
We should fail the getPartitions call if it exceeds the threshhold instead of return the results within limit.

Reply:
yes, the getPartition call for hive should fail in this case as well.

Reply:
For iceberg tables, we do not have the control and so we limit it to 100 in the case of no filter. So I added the same logic just for iceberg tables even when a filter is applied. If we want to apply the same in iceberg tables, then it should be done in iceberg api.

Reply:
Are you suggesting we use the same threshold for getPartitions as non-iceberg table and fail if we get back those number of partitions?

Reply:
We should let them know to handle the threshold case.